### PR TITLE
[TensorFlow] TF SM release

### PR DESCRIPTION
**Pull request by ohadkatz**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/training/docker/2.11/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+RUN ${PIP} uninstall -y tensorflow tensorflow-gpu \
+  ; ${PIP} install --no-cache-dir -U \
+    ${TF_URL} \
+    tensorflow-io==0.27.0 \
+    tensorflow-datasets==4.7.0
+
+# Install Horovod with TensorFlow
+RUN HOROVOD_WITH_MPI=1 \
+   && HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITHOUT_MXNET=1 HOROVOD_WITHOUT_PYTORCH=1 \
+   && ${PIP} install --no-cache-dir horovod==0.25.0
```
Comment (dkey-amazon) [inline tensorflow/training/docker/2.11/py3/Dockerfile.cpu:265]: This should be updated to be `horovod==0.26.0` to match the updated horovod version.

File: tensorflow/training/docker/2.11/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+   && rm -rf boost_1_73_0.tar.gz \
+   && rm -rf boost_1_73_0 \
+   && cd /usr/include/boost
+
+# add smdataparallel
+# smdataparallel
+# ARG SMDATAPARALLEL_BINARY=https://smdataparallel.s3.amazonaws.com/binary/tensorflow/2.10.0/cu112/2022-09-08/smdistributed_dataparallel-1.4.1-cp39-cp39-linux_x86_64.whl
+
+# Install SMD DP binary
+RUN SMDATAPARALLEL_TF=1 ${PYTHON} -m pip install --no-cache-dir ${SMDATAPARALLEL_BINARY}
```
Comment (dkey-amazon) [inline tensorflow/training/docker/2.11/py3/cu112/Dockerfile.gpu:405]: This line will fail since no `${SMDATAPARALLEL_BINARY}` value is specified.  Recommend commenting it out for now.

**Review by dkey-amazon — CHANGES_REQUESTED** [review]

File: tensorflow/training/docker/2.11/py3/cu112/Dockerfile.gpu (hunk 2 above)
Comment (ohadkatz) [reply tensorflow/training/docker/2.11/py3/cu112/Dockerfile.gpu:405]: Commented Out !

File: tensorflow/training/docker/2.11/py3/Dockerfile.cpu (hunk 1 above)
Comment (ohadkatz) [reply tensorflow/training/docker/2.11/py3/Dockerfile.cpu:265]: Updated to 0.26.0!
